services/server/src/server/cli.ts

```diff
@@ -23,6 +23,46 @@ import { SolcLocal } from "./services/compiler/local/SolcLocal";
 
 import session from "express-session";
 import { VyperLocal } from "./services/compiler/local/VyperLocal";
+import {
+  setLibSourcifyLoggerLevel,
+  SolidityMetadataContract,
+  SourcifyChain,
+} from "@ethereum-sourcify/lib-sourcify";
+
+// lib-sourcify configuration
+if (process.env.IPFS_GATEWAY || process.env.IPFS_GATEWAY_HEADERS) {
+  try {
+    SolidityMetadataContract.setGlobalIpfsGateway({
+      url: process.env.IPFS_GATEWAY || "https://ipfs.io/ipfs/",
+      headers: JSON.parse(process.env.IPFS_GATEWAY_HEADERS || "{}"),
+    });
+    logger.info("lib-sourcifs IPFS gateway set", {
+      ipfsGateway: process.env.IPFS_GATEWAY,
+      ipfsGatewayHeaders: process.env.IPFS_GATEWAY_HEADERS,
+    });
+  } catch (error) {
+    logger.error("Error setting lib-sourcify IPFS gateway", { error });
+    throw new Error("Error setting lib-sourcify IPFS gateway");
+  }
+}
+
+if (process.env.RPC_TIMEOUT) {
+  try {
+    SourcifyChain.setGlobalRpcTimeout(parseInt(process.env.RPC_TIMEOUT));
+    logger.info("lib-sourcify RPC timeout set", {
+      rpcTimeout: process.env.RPC_TIMEOUT,
+    });
+  } catch (error) {
+    logger.error("Error setting lib-sourcify RPC timeout", { error });
+    throw new Error("Error setting lib-sourcify RPC timeout");
+  }
+}
+
+if (process.env.NODE_ENV !== "production") {
+  // Set the log level to 4 (debug) in non-production environments
+  setLibSourcifyLoggerLevel(4);
+  logger.info("lib-sourcify log level set to debug");
+}
```

Comment:
This is how you actually set lib-sourcify's configs

Reply:
I didn't think about this! It's actually a good idea, let's pass them as Server's parameter!

Reply:
Notice that this variable should still be set in `cli.ts` because that's where we extract all the configs / env variables in order to pass them to `Server`.

Reply:
wdym? It's only dependent on the `NODE_ENV` variable. We can just look at that in Server and only set there. `cli` calls Server anyway. Why is it needed to be set in `cli`?

Reply:
I think it is weird that the log level is not consistent between lib-sourcify and server. Can we have one server constructor parameter that sets log level on server and lib-sourcify (and all other packages)?

Inside cli.ts we can set the value of this parameter from process.env.
Let's use this statement for setting it: 
```
const level = process.env.NODE_LOG_LEVEL || (process.env.NODE_ENV === "production" ? "info" : "debug");
 ```